File: server/app/api/v1/luyala_proxy/luyala_proxy.py
"""露雅拉代理 (外部AI API代理)"""


import httpx
from fastapi import APIRouter, Body
from loguru import logger

from app.schemas.common import error, success
from app.utils.ai_keys import luyala_key

router = APIRouter()

LUYALA_BASE_URL = "https://api.luyala.com/v1"


@router.post("/chat", operation_id="luyala_chat", summary="露雅拉对话")
async def chat(
    messages: list = Body(..., embed=True),
    model: str = Body("luyala-pro", embed=True),
    temperature: float = Body(0.7, embed=True),
    max_tokens: int = Body(2048, embed=True),
    api_key: str | None = None,
):
    async with httpx.AsyncClient(timeout=60) as client:
        try:
            headers = {"Content-Type": "application/json"}
            if luyala_key(api_key):
                headers["Authorization"] = f"Bearer {api_key}"
            r = await client.post(
                f"{LUYALA_BASE_URL}/chat/completions",
                json={
                    "model": model,
                    "messages": messages,
                    "temperature": temperature,
                    "max_tokens": max_tokens,
                },
                headers=headers,
            )
            return success(r.json())
        except Exception as e:
            logger.error(f"luyala chat error: {e}")
            return error(str(e))


@router.post("/completion", operation_id="luyala_completion", summary="露雅拉文本补全")
async def completion(
    prompt: str = Body(..., embed=True),
    model: str = Body("luyala-pro", embed=True),
    max_tokens: int = Body(1024, embed=True),
    api_key: str | None = None,
):
    async with httpx.AsyncClient(timeout=60) as client:
        try:
            headers = {"Content-Type": "application/json"}
            if luyala_key(api_key):
                headers["Authorization"] = f"Bearer {api_key}"
            r = await client.post(
                f"{LUYALA_BASE_URL}/completions",
                json={"model": model, "prompt": prompt, "max_tokens": max_tokens},
                headers=headers,
            )
            return success(r.json())
        except Exception as e:
            logger.error(f"luyala completion error: {e}")
            return error(str(e))


@router.post("/embeddings", operation_id="luyala_embeddings", summary="露雅拉Embedding")
async def embeddings(
    input_text: str = Body(..., embed=True),
    model: str = Body("luyala-embed", embed=True),
    api_key: str | None = None,
):
    async with httpx.AsyncClient(timeout=60) as client:
        try:
            headers = {"Content-Type": "application/json"}
            if luyala_key(api_key):
                headers["Authorization"] = f"Bearer {api_key}"
            r = await client.post(
                f"{LUYALA_BASE_URL}/embeddings",
                json={"model": model, "input": input_text},
                headers=headers,
            )
            return success(r.json())
        except Exception as e:
            logger.error(f"luyala embeddings error: {e}")
            return error(str(e))


@router.get("/models", operation_id="luyala_models", summary="可用模型列表")
async def models():
    try:
        async with httpx.AsyncClient(timeout=30) as client:
            r = await client.get(f"{LUYALA_BASE_URL}/models")
            return success(r.json())
    except Exception:
        return success(
            {
                "models": [
                    {"id": "luyala-pro", "name": "露雅拉Pro"},
                    {"id": "luyala-turbo", "name": "露雅拉Turbo"},
                    {"id": "luyala-embed", "name": "露雅拉Embed"},
                ]
            }
        )

File: client/src/api/luyala-proxy.ts
/**
 * 露雅拉代理 API
 * 对接后端: app/api/v1/luyala_proxy/luyala_proxy.py
 * 路由前缀: /api/v1/luyala-proxy
 *
 * 后端 chat/completion/embeddings 使用 Body(embed=True),
 * 即请求体为 JSON 对象, 字段名与后端一致 (snake_case)。
 */
import http from '@/utils/request'
import type { ApiResponse } from '@/types'

/** 露雅拉对话参数 */
export interface LuyalaChatParams {
  messages: Array<{ role: string; content: string }>
  model: string
  temperature?: number
}

/** 露雅拉文本补全参数 */
export interface LuyalaCompletionParams {
  prompt: string
  model: string
  maxTokens?: number
}

/** 露雅拉 Embedding 参数 */
export interface LuyalaEmbeddingParams {
  inputText: string
  model: string
  apiKey?: string
}

function toDataResult(data: unknown, msg = 'success'): ApiResponse<unknown> {
  return {
    code: 0,
    message: msg,
    data,
    success: true,
    timestamp: Date.now(),
  } as unknown as ApiResponse<unknown>
}

// ===========================================================================
// 露雅拉代理
// ===========================================================================

/** 露雅拉对话 (Body 传参) */
export async function luyalaChat(params: LuyalaChatParams): Promise<ApiResponse<unknown>> {
  const res = await http.post('/api/v1/luyala-proxy/chat', {
    messages: params.messages,
    model: params.model,
    temperature: params.temperature,
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg)
}

/** 露雅拉文本补全 (Body 传参) */
export async function luyalaCompletion(params: LuyalaCompletionParams): Promise<ApiResponse<unknown>> {
  const res = await http.post('/api/v1/luyala-proxy/completion', {
    prompt: params.prompt,
    model: params.model,
    max_tokens: params.maxTokens,
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg)
}

/** 露雅拉 Embedding (Body 传参, apiKey 走 Query) */
export async function luyalaEmbeddings(params: LuyalaEmbeddingParams): Promise<ApiResponse<unknown>> {
  const res = await http.post('/api/v1/luyala-proxy/embeddings', {
    input_text: params.inputText,
    model: params.model,
  }, {
    params: { api_key: params.apiKey || undefined },
  })
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg)
}

/** 可用模型列表 */
export async function luyalaModels(): Promise<ApiResponse<unknown>> {
  const res = await http.get('/api/v1/luyala-proxy/models')
  const body = (res as any).data || {}
  return toDataResult(body.data, body.msg)
}

export const luyalaProxyApi = {
  luyalaChat,
  luyalaCompletion,
  luyalaEmbeddings,
  luyalaModels,
}

export default luyalaProxyApi